src/lib/merkle_mask/maskable_merkle_tree.ml

```diff
@@ -22,28 +22,55 @@ module Make
 struct
   include Base
 
-  (* registered masks *)
-  let (mask_children : Mask.Attached.t list ref) = ref []
+  (* use an assoc list of (maskable, mask children) for registered masks
+
+     because of the types used in the test stubs for the databases in the ledger tests,
+     can't "derive hash" to allow using hash tables here, which would be a simpler implementation
+   *)
+
+  let (registered_masks : (t * Mask.Attached.t list) list ref) = ref []
```

Comment:
As discussed please rewrite with a hashtable (and we can fix the stubs)

Reply:
Done.